server/api/users.js

```diff
@@ -2,56 +2,82 @@ const router = require("express").Router();
 const {
   models: { User },
 } = require("../db");
-const { requireAdminToken, requireToken } =  require("../auth/index");
+const { requireAdminToken, requireToken } = require("../auth/index");
 
 module.exports = router;
 
 // GET all users -> /api/users
-// admin auth - TO BE INCLUDED
+// admin auth
 router.get("/", requireAdminToken, async (req, res, next) => {
   try {
     const users = await User.findAll({
       // explicitly select only the id and username fields - even though
       // users' passwords are encrypted, it won't help if we just
       // send everything to anyone who asks!
-      attributes: ["id", "username"],
+      attributes: [
+        "id",
+        "firstName",
+        "lastName",
+        "username",
+        "isAdmin",
+        "email",
+      ],
     });
     res.send(users);
   } catch (err) {
     next(err);
   }
 });
 
-// GET a single user -> /api/users/:id
-router.get("/:id", requireToken, async (req, res, next) => {
+// GET a single user -> /api/users/:userId
+router.get("/:userId", requireToken, async (req, res, next) => {
```

Comment:
when would this route be necessary if router.get("/me") in the auth server already does this? Could this route possible be used to browse through other user's profiles, in which case would we just take out the if statement in the body?

Reply:
good point, the /me function does already do this so I will remove it. In regards to your second comment, I think the If statement protects the endpoint so users wouldn't be able to view other user's data. Doesn't really matter now though lol.